### PR TITLE
Final fixes to make distributed wells work for Norne.

### PR DESCRIPTION
**Pull request by blattms**
This is tested manually and finally produces the correct result. Currenty, it is still not possible to request that wells might be cut  by process boundaries. I will prepare some more PRs for that later.

Current functionaility should remain unchanged, but there is no an additional copy if we need to access values
from perforations above. If that turns out to slow things down, then we can remove that copy for sequential runs. That would be some work.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                              | Relative |
| ------- | ------------------------------------------ | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1 | 1        |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8 | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1 | 1.003    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8 | 0.997    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1002

**Comment by blattms** [comment]
@atgeirr In case that was not clear: This was the PR that I meant when saying "quid pro quo" before Christmas.

File: opm/simulators/wells/StandardWell_impl.hpp [hunk 1]
```diff
@@ -2272,6 +2294,12 @@ namespace Opm
             connPI += np;
         }
 
+        //  Sum with communication in case of distributed well.
+        const auto& comm = this->parallel_well_info_.communication();
+        if (comm.size() > 1)
+        {
+            comm.sum(wellPI, np);
+        }
```
Comment (bska) [inline opm/simulators/wells/StandardWell_impl.hpp:2302]: Is there a reason you did not add this step to the [MultisegmentWell](https://github.com/OPM/opm-simulators/blob/aa38a2264d319e735aade14742f9e4897c13abe8/opm/simulators/wells/MultisegmentWell_impl.hpp#L1238-L1244) case?

Comment (blattms) [reply opm/simulators/wells/StandardWell_impl.hpp:2302]: Multisegment well is not treated, yet. Apart from the places where it was needed to satisfy necessary interface changes.

Comment (blattms) [reply opm/simulators/wells/StandardWell_impl.hpp:2302]: But they sure look very similar.

**Comment by atgeirr** [comment]
> @atgeirr In case that was not clear: This was the PR that I meant when saying "quid pro quo" before Christmas.

Thanks for the reminder, I'll look at it today afternoon.

**Comment by atgeirr** [comment]
I have read through the code and saw no problems. The only thing I want to ask is about perforation ordering. It was not quite clear to me what the global ordering will be. It is stated to be in the order given from the Schedule, but will that take into account the various possible ordering algorithms? I guess this is as much a question for @joakim-hove as for @blattms!

After reading this, I think we should perhaps make a tiny MPI help library to expand on what we get from Dune, to avoid having to do the "size and displ" dance over and over again, what do you think? (Not for this PR of course.)

I will rerun jenkins just in case, since it was run some time ago.

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
> It is stated to be in the order given from the Schedule, but will that take into account the various possible ordering algorithms? 

The `Schedule`ordering has taken the `COMPORD` keyword into account.

**Comment by atgeirr** [comment]
> The `Schedule`ordering has taken the `COMPORD` keyword into account.

Then all should be fine, thanks.

**Pull request merged**